pytrickle/stream_processor.py

```diff
@@ -319,8 +383,36 @@ async def process_audio_async(self, frame: AudioFrame) -> Optional[List[AudioFra
             return [frame]
     
     async def update_params(self, params: Dict[str, Any]):
-        """Update parameters using provided async function."""
-        if self.param_updater:
+        """Update parameters using provided async function.
+        
+        Handles the 'load_model' sentinel parameter internally by extracting it
+        and triggering model loading, while passing through all other parameters
+        to the user's param_updater callback.
+        """
+        # Handle model loading sentinel message (extract it with pop)
+        if params.pop("load_model", False):
```

Comment:
What is the workflow here? Is this what would be an indication the Client thinks this will require a __significant__ update to the pipeline that will take some time?  

If it is, what do you think of using `reload_pipeline`? This could possibly be loading a lora, changing pipeline configs only available at load time, etc.

Reply:
I like that idea of `reload_pipeline`. We can implement it as a managed state control `OK/IDLE` -> `LOADING` -> `OK/IDLE` which runs `load_model` and `warmup` via prompt update. pytrickle is sort of "model unaware" so I don't think we should do anything like force free gpu memory. IMHO it should be up to the consuming app to track and manage state of their model and to handle callbacks to `load_model` when their model is already loaded into memory appropriately. wdyt? @ad-astra-video

ComfyStream is unique in this because it automatically loads/unloads VRAM based on workflow changes.